File: inpyjama-30-days-of-rust/day11/day11.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main () {

   vector<string> treasure_map  = {"gold", "diamonds", "ruby"};
   vector<string> another_map = treasure_map ;
   another_map[0] = "shiny new sword";

   cout << "treasure_map : " << treasure_map[0]  << endl;
   cout << "another_map : " << another_map[0] << endl;


   return 0;
}

File: inpyjama-30-days-of-rust/day11/day11.py
# Python code
treasure_map = ["gold", "diamonds", "ruby"]
another_map = treasure_map

# Modifying another_map changes the original! (Shallow Copy)
another_map[0] = "shiny new sword"

print(treasure_map)  # Output: ["shiny new sword", "diamonds", "ruby"]
